substrate/frame/contracts/uapi/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! External C API to communicate with substrate contracts runtime module.
+//!
+//! Refer to substrate FRAME contract module for more documentation.
+
+#![no_std]
+
+mod flags;
+pub use flags::*;
+
+#[cfg(any(target_arch = "wasm32", target_arch = "riscv32"))]
+mod host;
+
+#[cfg(any(target_arch = "wasm32", target_arch = "riscv32"))]
+pub use host::*;
```

Comment:
Nit: Can we de-duplicate those attributes somehow?

Reply:
if we bring back cfg_if, not sure it's worth it for just saving this one line

Reply:
Yeah not worth it then.